kotlin/src/main/kotlin/org/axonframework/extensions/kotlin/serialization/AxonSerializers.kt

```diff
@@ -0,0 +1,346 @@
+import kotlinx.serialization.modules.polymorphic
+import kotlinx.serialization.modules.subclass
+import org.axonframework.eventhandling.GapAwareTrackingToken
+import org.axonframework.eventhandling.GlobalSequenceTrackingToken
+import org.axonframework.eventhandling.MergedTrackingToken
+import org.axonframework.eventhandling.MultiSourceTrackingToken
+import org.axonframework.eventhandling.ReplayToken
+import org.axonframework.eventhandling.TrackingToken
+import org.axonframework.eventhandling.scheduling.ScheduleToken
+import org.axonframework.eventhandling.scheduling.java.SimpleScheduleToken
+import org.axonframework.eventhandling.scheduling.quartz.QuartzScheduleToken
+import org.axonframework.eventhandling.tokenstore.ConfigToken
+import org.axonframework.messaging.responsetypes.InstanceResponseType
+import org.axonframework.messaging.responsetypes.MultipleInstancesResponseType
+import org.axonframework.messaging.responsetypes.OptionalResponseType
+import org.axonframework.messaging.responsetypes.ResponseType
+import kotlin.reflect.KClass
+
+private val trackingTokenSerializer = PolymorphicSerializer(TrackingToken::class).nullable
+
+val AxonSerializersModule = SerializersModule {
+    contextual(ConfigToken::class) { ConfigTokenSerializer }
+    contextual(GapAwareTrackingToken::class) { GapAwareTrackingTokenSerializer }
+    contextual(MultiSourceTrackingToken::class) { MultiSourceTrackingTokenSerializer }
+    contextual(MergedTrackingToken::class) { MergedTrackingTokenSerializer }
+    contextual(ReplayToken::class) { ReplayTokenSerializer }
+    contextual(GlobalSequenceTrackingToken::class) { GlobalSequenceTrackingTokenSerializer }
+    polymorphic(TrackingToken::class) {
+        subclass(ConfigTokenSerializer)
+        subclass(GapAwareTrackingTokenSerializer)
+        subclass(MultiSourceTrackingTokenSerializer)
+        subclass(MergedTrackingTokenSerializer)
+        subclass(ReplayTokenSerializer)
+        subclass(GlobalSequenceTrackingTokenSerializer)
+    }
+
+    contextual(SimpleScheduleToken::class) { SimpleScheduleTokenSerializer }
+    contextual(QuartzScheduleToken::class) { QuartzScheduleTokenSerializer }
+    polymorphic(ScheduleToken::class) {
+        subclass(SimpleScheduleTokenSerializer)
+        subclass(QuartzScheduleTokenSerializer)
+    }
+
+    contextual(InstanceResponseType::class) { InstanceResponseTypeSerializer }
+    contextual(OptionalResponseType::class) { OptionalResponseTypeSerializer }
+    contextual(MultipleInstancesResponseType::class) { MultipleInstancesResponseTypeSerializer }
+    contextual(ArrayResponseType::class) { ArrayResponseTypeSerializer }
+    polymorphic(ResponseType::class) {
+        subclass(InstanceResponseTypeSerializer)
+        subclass(OptionalResponseTypeSerializer)
+        subclass(MultipleInstancesResponseTypeSerializer)
+        subclass(ArrayResponseTypeSerializer)
+    }
+}
+
+object ConfigTokenSerializer : KSerializer<ConfigToken> {
```

Comment:
Question: Would a user of the Kotlin Extension ever be required to work with the `KSerializer` implementations in this file directly?